### PR TITLE
Add option to delete comments

### PR DESCRIPTION
**Pull request by mivdittamo**
- Add button that deletes all comments

- Add new servlet that has the doPost() method that goes to delete the comments in datastore

- Set minimum value for user input of the maximum number of comments to be displayed

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query(ENTITY_COMMENT);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity: results.asIterable()) {
+      Key entityKey = entity.getKey();
+      datastore.delete(entityKey);
+    }
+  }
+
```
Comment (estep-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java]: nit: remove line

File: portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
```
Comment (estep-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java]: nit: add newline at end of file

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -64,4 +64,11 @@ function createPElement(text) {
...
   pElement.innerText = text;
   return pElement;
+}
+
+function deleteComments() {
+  const request = new Request('/delete-data', {method: 'POST'});
+  fetch(request).then(response => {
+    getComments();
+  });
 }
```
Comment (estep-google) [inline portfolio/src/main/webapp/script.js]: nit: add newline at end of file

**Review by estep-google — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java (hunk 1 above)
Comment (mivdittamo) [reply portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java]: Got it!

File: portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java (hunk 2 above)
Comment (mivdittamo) [reply portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java]: Got it!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (mivdittamo) [reply portfolio/src/main/webapp/script.js]: Got it!

File: portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+
```
Comment (estep-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java:28]: nit: sorry I missed this early, remove this line as well (Java pretty much never uses two blank lines between anything)

Comment (mivdittamo) [reply portfolio/src/main/java/com/google/sps/servlets/DeletingDataServlet.java:28]: Got it!

**Review by estep-google — APPROVED** [review]
lgtm

**Pull request merged**